### PR TITLE
Fix #4622: Android Wiki migration

### LINKED ISSUE #4622: Migrate Android wiki to store source files in oppia-android repository

### PR DESCRIPTION
**Pull request by gp201**
## Explanation

- Fixes #4622
- The PR does the following: This PR moves the android wiki into the oppia-andoird codebase allowing contributors to make PRs to the wiki along with changes to the codebase itself. This PR also introduces a GitHub Action (Runs when a PR is merged into develop and when a user makes change to the wiki via the web interface) that copies the content of the wiki folder along with the git history of the wiki folder and deploys it to the oppia-android wiki repository.

The steps followed to achieve this wiki migration are present here: 
[Oppia Android Wiki Migration V2 (1).pdf](https://github.com/oppia/oppia-android/files/10921370/Oppia.Android.Wiki.Migration.V2.1.pdf)

### Note: Please ensure GitHub actions have write permissions. Please follow these instructions to enable it: 
[GrantingReadandWritePermissionsonGitHub_PDF.pdf](https://github.com/oppia/oppia-android/files/10919834/GrantingReadandWritePermissionsonGitHub_PDF.pdf)

## Essential Checklist

- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only

If your PR includes UI-related changes, then:
- Add screenshots for portrait/landscape for both a tablet & phone of the before & after UI changes
- For the screenshots above, include both English and pseudo-localized (RTL) screenshots (see [RTL guide](https://github.com/oppia/oppia-android/wiki/RTL-Guidelines))
- Add a video showing the full UX flow with a screen reader enabled (see [accessibility guide](https://github.com/oppia/oppia-android/wiki/Accessibility-(A11y)-Guide))
- Add a screenshot demonstrating that you ran affected Espresso tests locally & that they're passing

### CODE REVIEW

**Comment by gp201** [comment]
@BenHenning PTAL

File: .github/workflows/wiki.yml
Comment (BenHenning) [inline .github/workflows/wiki.yml:10]: What does this do?

File: .github/workflows/wiki.yml
Comment (BenHenning) [inline .github/workflows/wiki.yml]: Is the double ``--`` needed here?

File: wiki/_Sidebar.md
Comment (BenHenning) [inline wiki/_Sidebar.md]: Is using the ``[[`` syntax still correct here?

File: wiki/Internationalization.md
Comment (BenHenning) [inline wiki/Internationalization.md]: Ditto in this file--the ``[[`` doesn't seem to be rendering correctly.

File: wiki/Contributing-to-Oppia-Android.md
Comment (BenHenning) [inline wiki/Contributing-to-Oppia-Android.md]: Ditto in this file--the ``[[`` doesn't seem to be rendering correctly.

**Review by BenHenning — CHANGES_REQUESTED** [review]
Thanks @gp201! This looks fantastic! I appreciate your patience in me reviewing this.

I've verified that only a few changes were made between the original wiki files & these, so my comments are based more on how we should be formatting/keeping these files long-term. I think there might be a few small adjustments needed to make sure the files render correctly both when viewed directly & through the wiki.

I'm also wondering: would we benefit from some sort of READ_THIS_FIRST--type file in the wiki folder to explain how & when to change those files? Curious if you have any thoughts on this, as well, @seanlip.

**Comment by seanlip** [comment]
@BenHenning On the Oppia Web wiki page, we have a footer which explains how to edit the wiki: https://github.com/oppia/oppia/wiki/_Footer/_edit. This message shows up at the bottom of each page on the wiki. Is something like this what you were thinking?

File: .github/workflows/wiki.yml
Comment (gp201) [reply .github/workflows/wiki.yml:10]: This triggers the workflow when an wiki edit is made via the web interface. ([More information](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#gollum))

File: .github/workflows/wiki.yml
Comment (gp201) [reply .github/workflows/wiki.yml]: Yes, this command will be used to rewrite the repository to look as if `wiki/` had been its project root, and discard all other history. This step is only done on GitHub actions and deployed to the wiki repo. You can find more information [here](https://git-scm.com/docs/git-filter-branch#:~:text=To%20rewrite%20the%20repository%20to%20look%20as%20if%20foodir/%20had%20been%20its%20project%20root%2C%20and%20discard%20all%20other%20history%3A)

File: wiki/_Sidebar.md
Comment (gp201) [reply wiki/_Sidebar.md]: Yes, we cannot preview it in the PR. However, once it is deployed on the wiki it is viewable.

File: wiki/Internationalization.md
Comment (gp201) [reply wiki/Internationalization.md]: *Ditto.

File: wiki/Contributing-to-Oppia-Android.md
Comment (gp201) [reply wiki/Contributing-to-Oppia-Android.md]: *Ditto

**Comment by gp201** [comment]
@BenHenning PTAL

File: wiki/_Sidebar.md
Comment (BenHenning) [reply wiki/_Sidebar.md]: Do we need to use that syntax here & elsewhere? I suspect it'd be less confusing for the pages to render correctly both when viewed directly on GitHub & when on the wiki (though this matters less for the sidebar, but it's also harder to verify the links are correct during review time).

File: .github/workflows/wiki.yml
Comment (BenHenning) [reply .github/workflows/wiki.yml]: Ah, I think per the example this is supposed to be ``-- --all`` (without the extra space). It seems git is permissive here, but perhaps we should update the usage to more closely match the documentation?

File: .github/workflows/wiki.yml
Comment (BenHenning) [reply .github/workflows/wiki.yml:10]: Ah, thanks! Could you add a brief comment here to help contextualize this, e.g.: ``# Run this workflow when the wiki is changed (see https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#gollum).`` or something similar?

**Comment by BenHenning** [comment]
> @BenHenning On the Oppia Web wiki page, we have a footer which explains how to edit the wiki: https://github.com/oppia/oppia/wiki/_Footer/_edit. This message shows up at the bottom of each page on the wiki. Is something like this what you were thinking?

This would be excellent to have! That being said, I was actually thinking from the other direction (someone who sees the wiki/ directory is unsure how to change it). Perhaps a README.md in the wiki/ directory with a link that points to the wiki contribution guidelines should be sufficient (similar to the footer). Unless I'm mistaken, I think the readme will auto-render when viewing on GitHub so it helps provide some basis for folks. Now that you mention it, I think both are necessary here to make sure people understand how to change things correctly.

The issue I'm talking about might be unique to Android, too, since our wiki is alongside the code vs. in a different repository altogether. WDYT @gp201 @seanlip?

**Comment by seanlip** [comment]
Having a README in the wiki root, as well as a footer in the wiki explaining how to edit pages similar to Oppia web, seems fine to me. Thanks!

File: .github/workflows/wiki.yml
Comment (gp201) [reply .github/workflows/wiki.yml:10]: Done

File: .github/workflows/wiki.yml
Comment (gp201) [reply .github/workflows/wiki.yml]: Ah, nice catch! Fixed

File: wiki/_Sidebar.md
Comment (gp201) [reply wiki/_Sidebar.md]: We're working on a GitHub Action to lint check such Mediawiki links: https://github.com/oppia/oppia-web-developer-docs/issues/21
you can instead use the traditional URLs `[Link Text](full-URL-of-wiki-page)` if you prefer the pages to be rendered.

Comment (seanlip) [reply wiki/_Sidebar.md]: @gp201 but those don't work when looking at the .md file in GitHub code, right? E.g. see https://github.com/oppia/oppia-web-developer-docs/blob/develop/_Sidebar.md -- we might want to just use full links as a matter of course.

Comment (gp201) [reply wiki/_Sidebar.md]: ## How does GitHub render various links?
|Types of links				|GitHub Render																									|Raw text																|Syntax										|
|---						|---																											|---																	|---										|
|Mediawiki with a link text	|image|```[[Get involved!\|Home]]```													|`[[nameofwikipage\|Link-Text]]`				|
|Mediawiki with a URL link	|image|```[[Create a Lesson!\|https://github.com/oppia/oppia/wiki/Lesson-Creation-Guide]]```|`[[nameofwikipage\|full-URL-of-wiki-page]]`	|
|**Markdown Links**				|image|`[Oppia's mission](https://github.com/oppia/oppia/wiki/Oppia's-Mission)`	|`[Link Text](full-URL-of-wiki-page)`|

You can use Markdown Link syntax if you prefer the links to be rendered correctly.

## How to see how the page will render in wiki?
image
These options present at top right corner of every file in the PRs

- `Left option`: **Display the source diff**: This will allow you to see the raw markdown text
- `Right option`: **Display the rich diff**: This will allow you to see how the page will render in wiki

Comment (gp201) [reply wiki/_Sidebar.md]: Migrated all Media wiki links to markdown links

**Comment by gp201** [comment]
@U8NWXD Can you please take a look at the [wiki.md file](https://github.com/oppia/oppia-android/pull/4893/files#diff-89aca55dae2a3071993b5aff5e99906fb0804443dd94bf93b1de88b0453790a7) on behalf of the welfare team?

**Comment by seanlip** [comment]
Also adding @MohitGupta121 to take a look at Wiki.md: https://github.com/oppia/oppia-android/pull/4893/files#diff-89aca55dae2a3071993b5aff5e99906fb0804443dd94bf93b1de88b0453790a7

@BenHenning PTAL as codeowner. Thanks!

File: wiki/Wiki.md
Comment (MohitGupta121) [inline wiki/Wiki.md]: Replace the `open an issue` link from oppia-web to oppia-android

File: wiki/Wiki.md
Comment (MohitGupta121) [inline wiki/Wiki.md:18]: Can we also add here the Fork instructions? If some contributor came for docs contribution.

File: wiki/Wiki.md
Comment (MohitGupta121) [inline wiki/Wiki.md:37]: I think we need to add `style guide` as per oppia-android.
But for now It's good to have oppia-web.
@seanlip @BenHenning @gp201  What's your suggestions?

**Review by MohitGupta121 — CHANGES_REQUESTED** [review]
@gp201 LGTM, but some minor changes.

File: wiki/Wiki.md
Comment (gp201) [reply wiki/Wiki.md:37]: We aim to maintain the same style guide across both repos.
Can you elucidate how the oppia-android style guide may differ from that of oppia-web?

File: wiki/Wiki.md
Comment (gp201) [reply wiki/Wiki.md:18]: Do we expect the user to setup oppia-android repo or just fork and start contribution?

**Comment by gp201** [comment]
@MohitGupta121 I replied to some of your comments.

File: wiki/Wiki.md
Comment (MohitGupta121) [reply wiki/Wiki.md:18]: If he came for contributing to wiki then I think fork other wise he setup whole oppia-android repo.
But it's fine mostly user came for contributing to codebase.

File: wiki/Wiki.md
Comment (MohitGupta121) [reply wiki/Wiki.md:37]: I think something for UI related wiki we need to mention screenshots guide. Like for dark-mode.
What you think?

**Comment by MohitGupta121** [comment]
@gp201 Thanks, PTAL, replied to comments.

File: wiki/Wiki.md
Comment (gp201) [reply wiki/Wiki.md]: Done

File: wiki/Wiki.md
Comment (gp201) [reply wiki/Wiki.md:18]: Done

File: wiki/Wiki.md
Comment (gp201) [reply wiki/Wiki.md:37]: I think the screenshots guide can be used by both repos. Maybe if we ever see a user-case that requires us to create a separate style guide then we can make one for Oppia-android. What do you think?

**Comment by gp201** [comment]
@MohitGupta121 PTAL

File: wiki/Wiki.md
Comment (MohitGupta121) [reply wiki/Wiki.md:37]: Yes that's looks good to me, we can mame separate guide. If something more specific we need. For now it's good to have oppia-web. Thanks.

**Review by MohitGupta121 — APPROVED** [review]
@gp201 LGTM, Thanks.

File: wiki/Wiki.md
Comment (U8NWXD) [inline wiki/Wiki.md]: > Like on oppia/oppia-android

This suggests that the wiki is not on oppia/oppia-android, but I think it is, right?

> Note that we don't have code owners in the source repository.

There isn't a separate "source repository" is there? Since the oppia-android repo has code owners, why not just make the welfare team own everything under `wiki/`?

File: wiki/Wiki.md
Comment (U8NWXD) [inline wiki/Wiki.md]: I think "source repository" is confusing since it's not its own repository. This is a directory in the oppia-android repository

Also, shouldn't it be `/wiki` not `\wiki`? Ditto elsewhere

**Review by U8NWXD — CHANGES_REQUESTED** [review]

File: wiki/Wiki.md
Comment (BenHenning) [inline wiki/Wiki.md]: Could we just link to the existing set up instructions, instead? Prefer not to duplicate them in case one has to change.

File: wiki/Wiki.md
Comment (BenHenning) [reply wiki/Wiki.md]: I think the "source repository" terminology applies everywhere in this page, as well.

File: wiki/Wiki.md
Comment (BenHenning) [reply wiki/Wiki.md:37]: I think having something like a style guide makes sense, but I'm also not sure how exactly the rules/patterns are determined. This might become more clear (to me at least) with more PRs in the future changing the wiki (since, at the moment, most of the wiki pages have been created/edited by a small number of individuals so consistency hasn't been as much of an issue).

File: wiki/_Sidebar.md
Comment (BenHenning) [reply wiki/_Sidebar.md]: Thanks @gp201!

**Review by BenHenning — APPROVED** [review]
Thanks @gp201! The PR broadly LGTM, so approving it. I'm fine with @seanlip resolving any open comments from me once they're resolved, and merging.

@seanlip let's go with a squash merge here (per our discussion about whether to do a special one-time direct merge). I think it avoids introducing inconsistency in the history of the develop branch (since all other commits were proper PR squashes), and we'll have better wiki consistency moving forward as wiki files will be changed within PRs.

**Review by seanlip — CHANGES_REQUESTED** [review]
Putting a review hold since I'm taking over approval for Ben (just making sure his comments are resolved). @gp201 please ping me once they are and I'll merge it. Thanks!

File: wiki/Wiki.md
Comment (gp201) [reply wiki/Wiki.md]: Fixed

File: wiki/Wiki.md
Comment (gp201) [reply wiki/Wiki.md]: Fixed

File: wiki/Wiki.md
Comment (gp201) [reply wiki/Wiki.md]: Fixed

**Comment by gp201** [comment]
@seanlip @U8NWXD PTAL

**Review by seanlip — APPROVED** [review]
LGTM. Thanks @gp201!

**Pull request merged**
